### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by pingchenchan**
## Description

Implementation of the Ligon Flow frontend according to the latest version of Figma Flow.

- Login Page:
  - Route: http://localhost:3000/login
  - Remove the Google login option. 
  - Excluded: Authentication
- Select Name Page:
  - Route: http://localhost:3000/pick-your-name
  - Implement a dropdown menu that displays hardcoded names and supports user typing to filter options.
  - Include a "Your name isn't listed" link. Clicking on it will open a dialog displaying some information.
  - Implement a Continue button. If a name is selected, navigate to the PIN page. If not, a Snackbar will pop up reminding the user to select a name. 
  - Excluded: Retrieving usernames from the server
- Enter PIN Page:
  - Route: http://localhost:3000/enter-pin
  - Implement a 4-digit PIN entry field. It only accepts digits and supports keyboard controls like right arrow, left arrow, and backspace. Support temporary visibility, after 1 second of input the PIN digit will be automatically hidden. A SnackbarAlert is displayed if the user enters a non-number in the PIN entry.
  - Include an "I forgot my password" link. Clicking on it will open a dialog displaying some information.
  - Implement a Continue button. If a 4-digit PIN is entered, navigate to the PIN page. If not, a Snackbar will pop up reminding the user to enter the PIN.
  - Include a "Back to name selection" link. Clicking on it will return to the previous page. 
  - Excluded: Connect to the server

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

PIT-109

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings
Screenshots
image
image
image

### CODE REVIEW

File: src/pages/authentication/EnterPinPage.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import React, { useCallback, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Typography, Button, Dialog, DialogTitle, DialogContent, DialogActions, Stack, Snackbar, Alert } from '@mui/material';
+import Logo from '../../components/Logo/Logo';
+import MinimalWrapper from '../../layout/MinimalLayout/MinimalWrapper';
+import PinInput from '../../pages/authentication/PinInput';
```
Comment (bospoort) [inline src/pages/authentication/EnterPinPage.tsx:6]:
```suggestion
import PinInput from 'PinInput';
```

File: src/pages/authentication/EnterPinPage.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  const navigate = useNavigate();
+
+  const handleNextClick = () => {
+    if (pin.every(p => p !== "")) {
+      //TODO: Implement the logic to check the PIN
+      navigate('/enter-pin');
+    } else {
+      setOpenSnackbar(true);
+    }
+
```
Comment (bospoort) [inline src/pages/authentication/EnterPinPage.tsx:22]: nit: remove additional white line

File: src/pages/authentication/EnterPinPage.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+          }}
+        >
+          Continue
+        </Button>
+        <Typography
+          variant="body2"
+          onClick={handlePrevClick}
+          sx={{ cursor: 'pointer', textAlign: 'center', marginTop: 2, textDecoration: 'underline' }}
+        >
+          Back to the name selection
```
Comment (bospoort) [inline src/pages/authentication/EnterPinPage.tsx:93]:
```suggestion
          Back to the name selection.
```

File: src/pages/authentication/EnterPinPage.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    if (pin.every(p => p !== "")) {
+      //TODO: Implement the logic to check the PIN
+      navigate('/enter-pin');
+    } else {
+      setOpenSnackbar(true);
+    }
+
+  };
+
+  const handlePrevClick = () => {
```
Comment (bospoort) [inline src/pages/authentication/EnterPinPage.tsx:25]:
```suggestion
  const handlePreviousClick = () => {
```

File: src/pages/authentication/EnterPinPage.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+          Enter Your PIN
+        </Typography>
+        <PinInput onPinChange={handlePinChange} />
+        <Typography
+          variant="body2"
+          color="primary"
+          onClick={handleDialogOpen}
+          sx={{ cursor: 'pointer', textAlign: 'center', marginTop: 6, paddingTop:'3.66px' }}
+        >
+          I forgot my password
```
Comment (bospoort) [inline src/pages/authentication/EnterPinPage.tsx:73]:
```suggestion
          I forgot my password.
```

File: src/pages/authentication/EnterPinPage.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        <Typography
+          variant="h3"
+          textAlign="center"
+          sx={{
+            height: '50px', 
+            lineHeight: '50px', 
+            marginBottom: 2,
+          }}
+        >
+          Enter Your PIN
```
Comment (bospoort) [inline src/pages/authentication/EnterPinPage.tsx:64]:
```suggestion
          Enter Your PIN.
```

File: src/pages/authentication/PickNamePage.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        <Typography
+          variant="h3"
+          textAlign="center"
+          sx={{
+            height: '50px', 
+            lineHeight: '50px', 
+            marginBottom: 2,
+          }}
+        >
+          Pick Your Name
```
Comment (bospoort) [inline src/pages/authentication/PickNamePage.tsx:60]:
```suggestion
          Pick Your Name.
```

**Review by bospoort — APPROVED** [review]
Beautiful. Mostly nits and style differences. Looks good to me.

File: src/pages/authentication/EnterPinPage.tsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+const EnterPinPage: React.FC = () => {
+  const [open, setOpen] = useState<boolean>(false);
+  const [pin, setPin] = useState<string[]>(() => Array(4).fill(''));
+  const [openSnackbar, setOpenSnackbar] = useState<boolean>(false);
+  const navigate = useNavigate();
+
+  const handleNextClick = () => {
+    if (pin.every(p => p !== "")) {
+      //TODO: Implement the logic to check the PIN
+      navigate('/enter-pin');
```
Comment (bospoort) [inline src/pages/authentication/EnterPinPage.tsx:18]: can we navigate to home page here?

Comment (pingchenchan) [reply src/pages/authentication/EnterPinPage.tsx:18]: Yes, it's typo. It should be navigate('/');

File: src/pages/authentication/PickNamePage.tsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+          }}
+        >
+          Continue
+        </Button>
+
+        <Dialog open={openDialog} onClose={handleDialogClose}>
+          <DialogTitle>Contact PH Admin</DialogTitle>
+          <DialogContent>
+            <Typography>
+              Please call the phone number 222-222-2222 or email example@com of a PH admin who can assist you.
```
[the remark this answers is not in the document]
Comment (pingchenchan) [reply src/pages/authentication/PickNamePage.tsx]: Great idea. I build a reusable ContactAdminDialog component and add ADMIN_EMAIL and ADMIN_PHONE_NUMBER to .env file

**Pull request merged**